pkgs/top-level/static.nix

```diff
@@ -161,14 +161,15 @@ in {
   };
   mkl = super.mkl.override { enableStatic = true; };
   nix = super.nix.override { withAWS = false; };
-  # openssl 1.1 doesn't compile
-  openssl = super.openssl_1_0_2.override {
-    static = true;
-
-    # Don’t use new stdenv for openssl because it doesn’t like the
-    # --disable-shared flag
-    stdenv = super.stdenv;
-  };
+  openssl = (super.openssl_1_1.override { static = true; }).overrideAttrs (o: {
+    # OpenSSL doesn't like the `--enable-static` / `--disable-shared` flags.
+    # Instead, there's a specific `no-shared` configure flag.
+    # See https://wiki.openssl.org/index.php/Compilation_and_Installation#Configure_Options
+    # for a comprehensive list.
+    configureFlags = (removeUnknownConfigureFlags o.configureFlags) ++ [
+      "no-shared"
```

Comment:
could we move this into the openssl expression and add it in case `static == true`

Reply:
Hey @FRidh, thanks for your review. That does make sense, as with all the other derivations.

I (obviously) missed that when porting over from my own overlay. Just pushed a new commit that moves the static configuration to the openssl derivation.